### PR TITLE
adding port to config file

### LINKED ISSUE #23: Add http server port to config file
Allow the http server listening port to be configured in the config file.

### PR DESCRIPTION
**Pull request by sdf010**
the http server now has the option to set port and has to be started by calling  the start() function. The app now get the port option fromt he config file and passes it to the http server. closes #23.

### CODE REVIEW

File: src/http/server.js [hunk 1]
```diff
@@ -2,14 +2,26 @@ const http = require('http');
...
-        const guild = urlparameters.query.state;
-        logger.info(token);
-        logger.info(guild);
+let lport = 8080;
+
+module.exports.setPort = (port) => {
+    if (typeof port === 'number' && port <= 65535) {
+        lport = port;
+    } else {
+        throw new RangeError('Invlaid Listening Port');
```
Comment (f-doot) [inline src/http/server.js]: Fix the typo here and maybe also log the port, something like ``RangeError(`Invalid listening port ${port}`); ``

File: app.js [hunk 2]
```diff
@@ -34,6 +34,10 @@ fs.readdir('./src/events/', (err, files) => {
     return true;
 });
 
+// load the httpServer
+httpServer.setPort(config.http.port);
```
Comment (f-doot) [inline app.js:38]: Might be worth actually passing the port into the module when it's created, not 100% sure on that though.

**Review by f-doot — CHANGES_REQUESTED** [review]
Overall looks fine, just a couple things

File: app.js (hunk 2 above)
Comment (sdf010) [reply app.js:38]: As discussed we will be keeping the setter for this. If anyone has a better idea of how to manage httpServer config please share.

**Review by f-doot — APPROVED** [review]

**Pull request merged**